analytics-kotlin/src/test/java/com/segment/analytics/main/AndroidLifecyclePluginTests.kt

```diff
@@ -305,6 +305,80 @@ class AndroidLifecyclePluginTests {
         }
     }
 
+    @Config(sdk = [22])
+    @Test
+    fun `track deep link when enabled sdk=22`() {
+        analytics.configuration.trackApplicationLifecycleEvents = false
+        analytics.configuration.trackDeepLinks = true
+        analytics.configuration.useLifecycleObserver = false
+        analytics.add(lifecyclePlugin)
+        val mockPlugin = spyk(TestRunPlugin {})
+        analytics.add(mockPlugin)
+
+        val referrer = Uri.parse("android-app:/com.package.app")
+
+        val mockIntent = mockk<Intent>()
+        every { mockIntent.data } returns Uri.parse("app://track.com/open?utm_id=12345&gclid=abcd&nope=")
+        val mockActivity = mockk<Activity>()
+        every { mockActivity.intent } returns mockIntent
+        every { mockActivity.referrer } returns referrer
+        val mockBundle = mockk<Bundle>()
+
+        // Simulate activity startup
+        lifecyclePlugin.onActivityCreated(mockActivity, mockBundle)
+
+        assertTrue(mockPlugin.ran)
+        val track = slot<TrackEvent>()
+        verify { mockPlugin.track(capture(track)) }
+        with(track.captured) {
+            assertEquals("Deep Link Opened", event)
+            assertEquals(buildJsonObject {
+                put("url", "app://track.com/open?utm_id=12345&gclid=abcd&nope=")
+                put("utm_id", "12345")
+                put("gclid", "abcd")
+                put("referrer", "android-app:/com.package.app")
+            }, properties)
+        }
+    }
+/*
+    Due to some complications b/w mockk and robolectric this test seems to be failing on the CI
```

Comment:
for context: this test is working on my local machine, but for some reason fails on the CI, which I believe is stemming from a problem between mockk and robolectric